Changelog for 0.6.0-beta (#865)
hashicorp/consul-terraform-sync: CHANGELOG.md

* Fix issue where Task Status API response's `provider` field was returning provider name instead of provider id, which includes provider name and alias [[GH-798](https://github.com/hashicorp/consul-terraform-sync/issues/798)]
* Fix issue where the main function for running the CTS binary was only using a logger with default settings [[GH-789](https://github.com/hashicorp/consul-terraform-sync/issues/789)]
* Fix issue where `syslog.enabled` was not automatically set to true when other `syslog` configs were configured [[GH-815](https://github.com/hashicorp/consul-terraform-sync/pull/815)]
* Fix issue where deleting task run events used a read-lock instead of write-lock [[GH-787](https://github.com/hashicorp/consul-terraform-sync/pull/787)]

DEPRECATIONS:
* **(Enterprise Only)** Deprecate `terraform_version` task configuration in favor of `terraform_cloud_workspace.terraform_version` task configuration [[GH-790](https://github.com/hashicorp/consul-terraform-sync/issues/790)]
* **(Enterprise Only)** Deprecate `license_path` configuration in favor of `license.path` configuration [[GH-810](https://github.com/hashicorp/consul-terraform-sync/issues/810)]

> * Deprecate starting up CTS with no command in favor of starting up CTS using the new `start` command [[GH-866](https://github.com/hashicorp/consul-terraform-sync/issues/866)]

Reviewer: ```suggestion
* Deprecate starting CTS as a daemon using no CLI command in favor of starting CTS as a daemon using the new `start` command [[GH-866](https://github.com/hashicorp/consul-terraform-sync/issues/866)]
```

Reviewer: @lornasong what do you think of this?

Reviewer 2: Looks good to me, thank you. Will commit suggestion, locally squash, and then force push

Reviewer: sounds good!
